python/ray/util/client/server/server.py

```diff
@@ -167,31 +167,64 @@ def Init(
 
     @_use_response_cache
     def KVPut(self, request, context=None) -> ray_client_pb2.KVPutResponse:
-        with disable_client_hook():
-            already_exists = ray.experimental.internal_kv._internal_kv_put(
-                request.key, request.value, overwrite=request.overwrite
-            )
+        try:
```

Comment:
It seems wired. We should be able to pop exception to the client side. Could you give this a look https://stackoverflow.com/questions/40998199/raising-a-server-error-to-the-client-with-grpc

Btw, I'm not sure why context here can be None. If it's None, does this mean it's not a rpc call?

Reply:
+1.

Reply:
Synced with @jjyao and it seems return_exception_in_context covered this.

Reply:
I'm thinking about whether there is a way to make the code not verbose as this. It seems decorator won't work straightforward.
@jjyao do you have some ideas?

Reply:
I'll leave this to @ckw017.

Reply:
Looks fine to me (this is already how Terminate is implemented: https://github.com/ray-project/ray/blob/260d00a83f2aa6b3df3e297e68ebe7f4ebb706a6/python/ray/util/client/server/server.py#L323-L347). 

I'll dig around and see if there's a cleaner way to handle this, I suspect we can just encode the exception here: https://github.com/ray-project/ray/blob/260d00a83f2aa6b3df3e297e68ebe7f4ebb706a6/python/ray/util/client/server/server.py#L86-L88. This can be handled in a separate PR though.